keanu-examples/bayesianMethodsForHackers/src/main/java/com/examples/TextMessaging.java

```diff
@@ -50,10 +50,10 @@ public static TextMessagingResults run() {
         BayesianNetwork net = new BayesianNetwork(textsForDay.getConnectedGraph());
         net.probeForNonZeroProbability(1000);
 
-        final int numSamples = 50000;
+        final int numSamples = 800;
```

Comment:
By the looks of things - this is failing on Travis: Perhaps we need to make sure this test is deterministic? Or increase the number of samples slightly?

Reply:
It already has a `KeanuRandom.setDefaultRandomSeed(1)`

Reply:
`KeanuRandom` uses `org.apache.commons.math3.random.MersenneTwister` under the hood. Would you expect these to generate different numbers on different platforms (given the same seed)?

Reply:
I confirmed that our random seeds work the same on all platforms with a new test `KeanuRandomTest#randomNumberGenerationIsPlatformIndependent`.

I added `VertexId.ID_GENERATOR.set(1);` to every test and now it passes.